db/seeds.rb

```diff
@@ -1,8 +1,113 @@
+                                   template: 'standard',
+                                   menu_name: 'Sign Owner Slide',
+                                   heading: 'Slide by Sign Owner',
+                                   subheading: 'My slide for my signs',
+                                   datetime: Time.zone.now)
+sign_owners_slide.create_activity(:create,
+                                  owner: sign_owner,
+                                  parameters: { name: sign_owners_slide.menu_name })
+sign_owner.slides << sign_owners_slide
+UpdateSlide.execute(sign_owners_slide, {sign_ids: [private_sign.id]}, sign_owner)
+UpdateSlide.execute(sign_owners_slide, {sign_ids: [public_sign.id]}, sign_owner)
+
+# Non Owner creates two slides and requests they be added to Owner's sign.
+unapproved_slide = Slide.create!(name: 'Unapproved Sign',
+                                 template: 'standard',
+                                 menu_name: 'MA',
+                                 heading: 'Mature Audiences Only')
+unapproved_slide.create_activity(:create,
+                                 owner: non_sign_owner,
+                                 parameters: { name: unapproved_slide.menu_name })
+non_sign_owner.slides << unapproved_slide
+non_sign_owner.save!
+UpdateSlide.execute(unapproved_slide, {sign_ids: [public_sign.id]}, non_sign_owner)
+
+approved_slide = Slide.create!(name: 'Approved Slide',
+                               template: 'standard',
+                               menu_name: 'Approved Slide',
+                               heading: 'A Publicly Approved Slide')
+approved_slide.create_activity(:create,
+                               owner: non_sign_owner,
+                               parameters: { name: approved_slide.menu_name })
+non_sign_owner.slides << approved_slide
+non_sign_owner.save!
+UpdateSlide.execute(approved_slide, {sign_ids: [public_sign.id]}, non_sign_owner)
+
+# Sign Owner approves approved sign.
+approved_slide.sign_slides.first.update(approved: true)
+public_sign.touch
+UserMailer.sign_slide_approved(sign_slide: approved_slide.sign_slides.first,
+                               approver: sign_owner,
+                               message: 'I approve this sign!').deliver_now
+
+# Non Owner create slide that has expired
+expired_slide = Slide.create!(name: 'Expired Slide',
+                              template: 'standard',
+                              menu_name: 'Expired Slide',
+                              heading: 'This slide died alone and unloved.',
+                              play_on: Time.zone.now - 5.days,
+                              stop_on: Time.zone.now - 2.days)
+expired_slide.create_activity(:create,
+                              owner: non_sign_owner,
+                              parameters: { name: approved_slide.menu_name })
+non_sign_owner.slides << expired_slide
+non_sign_owner.save!
+UpdateSlide.execute(expired_slide, {sign_ids: [public_sign.id]}, non_sign_owner)
```

Comment:
Example of a bad interface:

    UpdateSlide.execute(slide, params, user)

Better:

    user.approves_slide_for_sign(slide, sign)

Or:

    slide.approved_for(sign, owner)